odoo/models.py

```diff
@@ -5161,7 +5141,13 @@ def browse(self, ids=None):
             ids = (ids,)
         else:
             ids = tuple(ids)
-        return self._browse(self.env, ids, ids)
+        return self.__class__(self.env, ids, ids)
+
+    @classmethod
+    def _browse(cls, env, ids, prefetch_ids):
+        # backward compatibility
+        warnings.warn("_browse() is deprecated, use class instantiation instead", DeprecationWarning)
+        return cls(env, ids, prefetch_ids)
```

Comment:
who has any reason to call `_browse()` directly? that seems like a low-level internal thing that people shouldn't be using. I'd drop it rather than decreasing the signal-to-noise ratio of the framework code.